model/quic-socket-base.h

```diff
@@ -724,6 +724,8 @@ class QuicSocketBase : public QuicSocket
   TracedValue<Time> m_drainingPeriodTimeout;  //!< Draining Period timeout
   EventId m_sendAckEvent;                     //!< Send ACK timeout event
   EventId m_delAckEvent;                      //!< Delayed ACK timeout event
+  bool m_flushOnClose;                        //!< Control behavior on connection close
```

Comment:
Why do you need two variables?

Reply:
m_flushOnClose is the attribute that defines the behavior. If it is true, m_closeOnEmpty becomes true while the buffer is flushing and is used to check whether the transmission is over and it's safe to close the socket.